tests/util/models/gpflow/models.py

```diff
@@ -123,6 +139,32 @@ def mock_data() -> tuple[tf.Tensor, tf.Tensor]:
     )
 
 
+class QuadraticMeanAndRBFKernelWithSamplers(QuadraticMeanAndRBFKernel):
+    r"""
+    A Gaussian process with scalar quadratic mean, an RBF kernel and
+    a trajectory_sampler and reparam_sampler methods.
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        *,
+        x_shift: float | SequenceN[float] | TensorType = 0,
+        kernel_amplitude: float | TensorType | None = None,
+        noise_variance: float = 1.0,
+    ):
+        super().__init__(
+            x_shift=x_shift, kernel_amplitude=kernel_amplitude, noise_variance=noise_variance
+        )
+        self._dataset = dataset
+
+    def trajectory_sampler(self) -> TrajectorySampler:
+        return RandomFourierFeatureTrajectorySampler(self, self._dataset, 100)
+
+    def reparam_sampler(self, num_samples: int) -> ReparametrizationSampler:
+        return BatchReparametrizationSampler(num_samples, self)
```

Comment:
(might it be useful to have this use IndependentReparametrizationSampler instead or does that not make sense?)

Reply:
This needs to be the batch one, e.g. for batch Monte-Carlo EI